File: replace-plugin/src/main/java/com/maple/replace/replace/ReplaceConfig.java
package com.maple.replace.replace;

import org.objectweb.asm.Opcodes;

import java.util.ArrayList;
import java.util.List;

public class ReplaceConfig {

    public static List<ReplaceBean> getReplaceBeans() {
        List<ReplaceBean> beans = new ArrayList<>();
        beans.add(new ReplaceBean(Opcodes.INVOKEVIRTUAL,
                "android/content/Context",
                "sendBroadcast",
                "(Landroid/content/Intent;)V",
                false)
                .setNewCodeConfig(Opcodes.INVOKESTATIC,
                        "com/gavin/asmdemo/BroadcastUtils",
                        "sendAppInsideBroadcast",
                        "(Landroid/content/Context;Landroid/content/Intent;)V",
                        false)
        );
        return beans;
    }

}

File: base-plugin/src/main/kotlin/com/maple/plugin/plugins/replace/ReplaceConfig.kt
package com.maple.plugin.plugins.replace

import org.objectweb.asm.Opcodes

object ReplaceConfig {

    @JvmStatic
    fun getReplaceBeans(): List<ReplaceBean> {
        return listOf(
            ReplaceBean(
                Opcodes.INVOKEVIRTUAL,
                "android/content/Context",
                "sendBroadcast",
                "(Landroid/content/Intent;)V",
                false
            ).setNewCodeConfig(
                Opcodes.INVOKESTATIC,
                "com/maple/asm_learn/BroadcastUtils",
                "sendAppInsideBroadcast",
                "(Landroid/content/Context;Landroid/content/Intent;)V",
                false
            )
        )
    }
}